mineinabyss-features/src/main/kotlin/com/mineinabyss/playerprofile/PlayerProfileMenu.kt

```diff
@@ -0,0 +1,198 @@
+package com.mineinabyss.playerprofile
+
+import androidx.compose.runtime.Composable
+import com.mineinabyss.components.playerData
+import com.mineinabyss.components.players.Patreon
+import com.mineinabyss.geary.papermc.access.toGeary
+import com.mineinabyss.guilds.extensions.*
+import com.mineinabyss.guilds.menus.GuildScreen
+import com.mineinabyss.guiy.components.Item
+import com.mineinabyss.guiy.components.canvases.Chest
+import com.mineinabyss.guiy.inventory.GuiyOwner
+import com.mineinabyss.guiy.inventory.guiy
+import com.mineinabyss.guiy.layout.Column
+import com.mineinabyss.guiy.modifiers.Modifier
+import com.mineinabyss.guiy.modifiers.at
+import com.mineinabyss.guiy.modifiers.height
+import com.mineinabyss.helpers.*
+import com.mineinabyss.helpers.ui.composables.Button
+import com.mineinabyss.idofront.font.Space
+import com.mineinabyss.idofront.messaging.miniMsg
+import net.luckperms.api.LuckPermsProvider
+import net.luckperms.api.node.NodeType
+import net.luckperms.api.node.types.InheritanceNode
+import org.bukkit.Bukkit
+import org.bukkit.ChatColor.*
+import org.bukkit.Statistic
+import org.bukkit.entity.Player
+
+
+val luckPerms = LuckPermsProvider.get()
+
+@Composable
+fun GuiyOwner.PlayerProfile(viewer: Player, player: Player) {
+    val isPatreon = player.toGeary().has<Patreon>()
+    val titleName = player.name.toList().joinToString { ":player_profile_$it:" }.replace(", ", "")
+    val ranks = DisplayRanks(player)
+
+    Chest(setOf(viewer),
+        "${Space.of(-12)}:player_profile${if (isPatreon) "_patreon:" else ":"}${Space.of(-178)}$titleName${Space.of(-42)}${ranks}",
+        Modifier.height(4),
+        onClose = { viewer.closeInventory() }) {
+        PlayerHead(player, Modifier.at(0, 1))
+        Column(Modifier.at(2, 0)) {
+            //DisplayRanks(player)
+        }
+        Column(Modifier.at(5, 0)) {
+            OrthCoinBalance(player)
+            if (isPatreon) CloutTokenBalance(player)
+            GuildButton(player, viewer)
+            DiscordButton(player)
+        }
+        Column(Modifier.at(7, 0)) {
+            CosmeticHat(player)
+            CosmeticBackpack(player)
+        }
+        Column(Modifier.at(8, 0)) {
+            HelmetSlot(player)
+
+            ChestplateSlot(player)
+            LeggingsSlot(player)
+            BootsSlot(player)
+        }
+    }
+}
+
+@Composable
+fun PlayerHead(player: Player, modifier: Modifier) {
+    Item(
+        player.head(
+            "${LIGHT_PURPLE}${BOLD}${player.name}",
```

Comment:
I think we wanna switch to MiniMessage here and make all the functions auto parse each string